app/src/Components/AdminPage/VisitorsTableSignedOut/VisitorsTableSignedOut.js

```diff
@@ -0,0 +1,190 @@
+import React from "react";
+import '../VisitorsTableSignedIn/visitorsTable.css';
+import InfiniteScroll from "react-infinite-scroll-component";
+
+class VisitorsTableSignedOut extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            visitorPackage: [],
+            lastFetchedVisitors: [],
+            bearerToken: localStorage.getItem('bearerToken'),
+            appUrl: localStorage.getItem('appUrl'),
+            signedOutTableVisible: 'd-none',
+            endMessageVisibility: 'd-block',
+            hasMore: true
+        };
+    }
+
+    componentDidMount() {
+        this.initialTableRenderData()
+    }
+
+    initialTableRenderData = async () => {
+        const count = 15;
+        let firstBatch = await this.fetchVisitors(count, 999999);
```

Comment:
Is this done like this so that you don't have to make a fetch to grab all the records to count them? Potentially we could send a query to the database that simply returned a count of all the signed out records (rather than all the data) in order to get this number.

Reply:
such a good idea! I'll try and do this. You guess is correct. 999999 was the equivalent of having 10 visitors a day for 274 years... at which point, i figured they wouldn't be using the app anymore lol

Reply:
all done! wasn't able to simply grab the length count and use that as the start variable. This was because, as people sign out of the current day, the array total count will almost alway be less than the highest signout out visitor id. So, I just grabbed the id of the first in the array when requesting _all_ signed out visitors which always gave me the highest id to start from. Hope that makes sense lol